loom/graphics/gfxTexture.cpp

```diff
@@ -822,9 +870,78 @@ void Texture::reset()
     }
 }
 
+void Texture::clear(TextureID id, int color, float alpha)
+{
+	bool current = currentRenderTexture == id;
+
+	TextureID prevRenderTexture = currentRenderTexture;
+	
+	setRenderTarget(id);
+
+	Graphics::context()->glClearColor(
+		float((color >> 16) & 0xFF) / 255.0f,
+		float((color >> 8) & 0xFF) / 255.0f,
+		float((color >> 0) & 0xFF) / 255.0f,
+		alpha
+	);
+	Graphics::context()->glClear(GL_COLOR_BUFFER_BIT);
+
+	setRenderTarget(prevRenderTexture);
+}
+
+void Texture::setRenderTarget(TextureID id)
+{
+	if (id != -1)
+	{
+		if (currentRenderTexture == id) return;
+		setRenderTarget(-1);
+		lmAssert(currentRenderTexture == -1, "Internal setRenderTarget error, render already in progress");
+
+		currentRenderTexture = id;
+
+		id &= TEXTURE_ID_MASK;
```

Comment:
Yes, this is probably a good place too, so we don't accidentally set a render target to a recycled texture.